Fix registration for existing users

omarkhan opened the pull request:
I noticed a bug in the beta registration after I deployed it to stage. If you are already logged in, but have not yet registered, the form will not allow you to register ('username is already taken'). This pull request fixes the bug by ignoring the logged in user when checking for existing usernames and emails.

#### Testing

1. Create a new user in the admin or the shell
2. Log in as that user
3. Go to `/registration/`. The username and email fields should be prepopulated and read only, and the password fields should not appear.
4. Fill in and submit the form

e-kolpakov commented on registration/tests/browser/browser_registration.py:
```diff
@@ -60,18 +61,47 @@ def url(self):
+            except NoSuchElementException:
+                # Some form fields, like the password fields, only appear if
+                # the user is not already logged in.
+                continue
```
@omarkhan feels like a bad idea - shouldn't caller not pass field value in field_data in such case?

omarkhan replied:
Good point, fixed

e-kolpakov commented:
@omarkhan All good and works as specified, but CI build fails; so :+1: conditional build passes.

One more thing - it is not related to the bug you're fixing, but instance/#/isntances have some minor UI glitch - double logout buttons:

![image](https://cloud.githubusercontent.com/assets/3330048/15538683/635547b0-2286-11e6-9f96-b1ef9fced8aa.png)

Might be cool to fix it; or log a bug for later.

omarkhan commented:
Thanks for the tip about the double logout buttons - I had removed it in the last PR but Braden added it back in another template before I merged it. I have removed it again, I will merge when the build is green.
